src/esp/sim/Simulator.cpp

```diff
@@ -148,7 +149,7 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
   sceneID_.push_back(activeSceneID_);
 
   if (config_.createRenderer) {
-    if (!context_) {
+    if (!context_ && !Magnum::GL::Context::hasCurrent()) {
```

Comment:
I'd put some comment here (because even if I knew what was this for back then, I definitely don't remember now).

Reply:
It had something to do with the changes needed for the Python viewer on #727

Reply:
Yeah that's what I remember, but I don't remember *why* :sweat_smile: @dexter1691 do *you* remember? :)

Reply:
I think this was because a GL context already exists and it crashes since two magnum context cannot exist. When creating a viewer based app, there is no need to create a WindowlessContext since another context already exists. This would be required both for python viewer and this PR.

Reply:
Yeah, creating a Simulator instance will try to create another context.